client/state/reducer.js

```diff
@@ -16,6 +17,7 @@ import sites from './sites/reducer'; // eslint-disable-line no-restricted-import
 const reducers = {
 	currentUser,
 	dataRequests,
+	downgradeModal,
```

Comment:
I don't think this needs to live in a redux store. You can probably introduce a new hook inside plan-features-main which maintains this state And returns a flag and a callback.

client module Redux is generally discouraged nowadays unless its absolutely necessary. I cannot find the post but I remember @tyxla talking about this.

The main reason is the unnecessary boilerplate. For complex state management we use wordpress store.

Let me know if you want me to take over since I got to know from @southp that we are racing towards a tough deadline and there is more work to be done!

Reply:
Yep, that's the case unless we need to interact with the other existing Redux state trees.

We've been talking about this in many places, but this post and the discussion underneath should bring enough context:

p4TIVU-auD-p2

Reply:
Thanks for reviewing and bringing this up! I'll keep this in mind for now on.

Since the downgrsade modal is likely to be moved to /me/purchases, there's a chance the hook might not be necessary (p58i-kiA-p2#comment-67118).